src/estimate.c

```diff
@@ -51,8 +52,25 @@ estimate_max_spread_var(PlannerInfo *root, Var *var)
 	}
 	PG_CATCH();
 	{
-		valid = false;
+		MemoryContext ecxt = MemoryContextSwitchTo(ccxt);
+		ErrorData *edata = CopyErrorData();
+
+		/*
+		 * We can deal with the Data Exception category and in the Syntax
+		 * Error or Access Rule Violation category, but if the error is an
+		 * insufficient resources category, for example, an out of memory
+		 * error, we should just re-throw it and not treat it as an invalid
+		 * value.
+		 */
+		if (ERRCODE_TO_CATEGORY(edata->sqlerrcode) != ERRCODE_DATA_EXCEPTION &&
+			ERRCODE_TO_CATEGORY(edata->sqlerrcode) != ERRCODE_SYNTAX_ERROR_OR_ACCESS_RULE_VIOLATION)
+		{
+			FreeErrorData(edata);
+			MemoryContextSwitchTo(ecxt);
+			PG_RE_THROW();
+		}
 		FlushErrorState();
+		valid = false;
```

Comment:
AFAIK, it is not correct to continue processing here. See here for explanation: https://www.postgresql.org/message-id/27190.1508727890%40sss.pgh.pa.us

I really think we need to avoid this pattern in our code.

Reply:
Mostly want to avoid triggering the panic right now, so not doing too large a rewrite at this point, but followed the pattern of `exec_stmt_block` and started a sub-transaction to execute the functions.

Reply:
Unless we do a proper fix, we might just mask the problem. Like Tom Lane says in the thread: "If it works, it's only because you didn't try very hard to break it."

Using a subtransaction and rolling it back on error seems like a valid approach.

Reply:
> Using a subtransaction and rolling it back on error seems like a valid approach.

This is what the code does? Did you have another suggestion for how to do it?

Reply:
>This is what the code does? Did you have another suggestion for how to do it?

Yes, I was just acknowledging.

Reply:
Note that I think we should validate the input before going this deep into execution and even end up in a position where we need to capture errors, but this is a bigger change and something I want to avoid right now.